pkg/webui/console/views/device-add/validation-schema.js

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as Yup from 'yup'
+
+import sharedMessages from '../../../lib/shared-messages'
+import { id as deviceIdRegexp } from '../../lib/regexp'
+import m from './messages'
+
+export default Yup.object().shape({
+  ids: Yup.object().shape({
+    device_id: Yup.string()
+      .matches(deviceIdRegexp, sharedMessages.validateAlphanum)
+      .min(2, sharedMessages.validateTooShort)
+      .max(36, sharedMessages.validateTooLong)
+      .required(sharedMessages.validateRequired),
+  }).when('activation_mode', {
+    is: 'otaa',
+    then: Yup.object().shape({
+      join_eui: Yup.string()
+        .length(8 * 2, m.validate16)
+        .required(sharedMessages.validateRequired), // 8 Byte hex
+      dev_eui: Yup.string()
+        .length(8 * 2, m.validate16)
+        .required(sharedMessages.validateRequired), // 8 Byte hex
+    }),
+  }),
+  session: Yup.object().shape({
+    dev_addr: Yup.string().length(4 * 2, m.validate8), // 4 Byte hex
```

Comment:
this and the `keys` can be empty.
typing a couple of characters -> clearing up the field completely leaves the error under the field